### PR TITLE
Add AlphaRowsetMeta and AlphaRowsetReader

### PR DESCRIPTION
**Pull request by kangpinghuang**
1. Add AlphaRowsetMeta. Original PDelta will be converted to RowsetMetaPb and will be managed by AlphaRowsetMeta
2. Add AlphaRowsetReader. AlphaRowsetReader will replace original ColumnData and will be the interface to provide data. AlphaRowsetReader  will encapsulate original reading data logic.

### CODE REVIEW

**Review by yiguolei — APPROVED** [review]

File: be/src/olap/rowset/alpha_rowset_builder.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#ifndef DORIS_BE_SRC_OLAP_ROWSET_ALPHA_ROWSET_BUILDER_H
+#define DORIS_BE_SRC_OLAP_ROWSET_ALPHA_ROWSET_BUILDER_H
+
+#include "olap/rowset/rowset_builder.h"
+
+namespace doris {
+
+class AlphaRowsetBuilder {
+public:
+    virtual NewStatus init(std::string rowset_id, const std::string& rowset_path_prefix, Schema* schema);
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset_builder.h]: std::string -> int64_t

File: be/src/olap/rowset/alpha_rowset_builder.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+#include "olap/rowset/alpha_rowset_builder.h"
+
+namespace doris {
+
+NewStatus AlphaRowsetBuilder::init(std::string rowset_id, const std::string& rowset_path_prefix, Schema* schema) {
+    return NewStatus.OK();
+}
+
+NewStatus AlphaRowsetBuilder::add_row_block(RowBlock* row_block) {
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset_builder.cpp]: row_block will not be changed. May be use const RowBlock&

File: be/src/olap/rowset/alpha_rowset.cpp [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+std::unique_ptr<RowsetReader> AlphaRowset::create_reader() {
+    std::vector<SegmentGroup*> segment_groups;
+    for (auto& segment_group : _segment_groups) {
+        segment_groups.push_back(segment_group.get());
+    }
+    AlphaRowsetReader* rowset_reader = new AlphaRowsetReader(_tablet_schema,
+            _num_key_fields, _num_short_key_fields, _num_rows_per_row_block,
+            _rowset_path, _rowset_meta.get(), _segment_groups);
+    return std::unique_ptr<RowsetReader>(rowset_reader);
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset.cpp]: unique_ptr can be assigned?

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset/alpha_rowset_meta.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    return parsed;
+}
+
+void AlphaRowsetMeta::get_segment_groups(std::vector<PSegmentGroup>* segment_groups) {
+    for (auto& segment_group : _extra_meta_pb.segment_groups()) {
+        segment_groups->push_back(segment_group);
+    }
+}
+
+void AlphaRowsetMeta::add_segment_group(PSegmentGroup& segment_group) {
```
[the remark this answers is not in the document]
Comment (kangpinghuang) [reply be/src/olap/rowset/alpha_rowset_meta.cpp]: I will use const PSegmentGroup&, and I will add another patch to fix PSegmentGroup to SegmentGroupPB

File: be/src/olap/rowset/alpha_rowset.cpp (hunk 3 above)
Comment (kangpinghuang) [reply be/src/olap/rowset/alpha_rowset.cpp]: unique_ptr can be used as return value

File: be/src/olap/rowset/alpha_rowset_builder.cpp (hunk 2 above)
Comment (kangpinghuang) [reply be/src/olap/rowset/alpha_rowset_builder.cpp]: ok

File: be/src/olap/rowset/alpha_rowset_builder.h (hunk 1 above)
Comment (kangpinghuang) [reply be/src/olap/rowset/alpha_rowset_builder.h]: ok, I will fix it

**Pull request merged**